sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/AesEncryptionShouldBeUsedWithSecuredMode.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Security.Cryptography;
+
+namespace Tests.Diagnostics
+{
+    public class Program
+    {
+        public void CompliantCases()
+        {
+            // None
+        }
+
+        public void NonCompliantCases()
+        {
+            var aes = new AesManaged();
+
+            new AesManaged
+            {
+                Mode = CipherMode.ECB // Noncompliant
+//                     ^^^^^^^^^^^^^^
+            };
+
+            new AesManaged
+            {
+                Mode = CipherMode.CBC, // Noncompliant
+            };
+
+            new AesManaged
+            {
+                Mode = CipherMode.OFB // Noncompliant
+            };
+
+            new AesManaged
+            {
+                Mode = CipherMode.CFB // Noncompliant
+            };
+
+            new AesManaged
+            {
+                Mode = CipherMode.CTS // Noncompliant
+            };
+
+            aes.Mode = CipherMode.ECB; // Noncompliant
```

Comment:
Maybe add a test case to show we are only raising issues for modes we know about e.g.
 _aes.Mode = CipherMode.NewUnknownSecureMode;_

(it's possible that a future version of the framework might include a secure cipher mode)